pinot-common/src/main/java/org/apache/pinot/common/metadata/ZKMetadataProvider.java

```diff
@@ -317,6 +338,18 @@ public static TableConfig getTableConfig(ZkHelixPropertyStore<ZNRecord> property
         AccessOption.PERSISTENT));
   }
 
+  /**
+   * @return a pair of table config and current version from znRecord.
+   */
+  public static ImmutablePair<TableConfig, Integer> getTableConfigWithVersion(
+      ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType) {
+    Stat tableConfigStat = new Stat();
+    TableConfig tableConfig = toTableConfig(
+        propertyStore.get(constructPropertyStorePathForResourceConfig(tableNameWithType), tableConfigStat,
+            AccessOption.PERSISTENT));
+    return ImmutablePair.of(tableConfig, tableConfigStat.getVersion());
```

Comment:
When `tableConfig` is null, we should probably directly return `null` instead of a pair. Also annotate the return as `@Nullable`